pkg/process/subscribers/gpu_subscriber_test.go

```diff
@@ -0,0 +1,313 @@
+		},
+		{
+			name:           "Detected gpu with empty workloadmeta",
+			detectedGPU:    true,
+			gpus:           []workloadmeta.GPU{},
+			expectedTagMap: map[int32][]string{},
+		},
+		{
+			name:        "Detected process on gpu",
+			detectedGPU: true,
+			gpus: []workloadmeta.GPU{
+				{
+					EntityID: entityID,
+					EntityMeta: workloadmeta.EntityMeta{
+						Name: entityID.ID,
+					},
+					Vendor:     "nvidia",
+					Device:     "tesla-v100",
+					ActivePIDs: []int{1234},
+				},
+				{
+					EntityID: workloadmeta.EntityID{
+						Kind: workloadmeta.KindGPU,
+						ID:   "gpu-2",
+					},
+					EntityMeta: workloadmeta.EntityMeta{
+						Name: "gpu-2",
+					},
+					Vendor:     "nvidia",
+					Device:     "tesla-v105",
+					ActivePIDs: []int{185},
+				},
+			},
+			expectedTagMap: map[int32][]string{
+				1234: {
+					"gpu_uuid:gpu-1",
+					"gpu_device:tesla-v100",
+					"gpu_vendor:nvidia",
+				},
+				185: {
+					"gpu_uuid:gpu-2",
+					"gpu_device:tesla-v105",
+					"gpu_vendor:nvidia",
+				},
+			},
+		},
+		{
+			name:        "Detected process on multiple gpus",
+			detectedGPU: true,
+			gpus: []workloadmeta.GPU{
+				{
+					EntityID: entityID,
+					EntityMeta: workloadmeta.EntityMeta{
+						Name: entityID.ID,
+					},
+					Vendor:     "nvidia",
+					Device:     "tesla-v100",
+					ActivePIDs: []int{1234, 185},
+				},
+				{
+					EntityID: workloadmeta.EntityID{
+						Kind: workloadmeta.KindGPU,
+						ID:   "gpu-2",
+					},
+					EntityMeta: workloadmeta.EntityMeta{
+						Name: "gpu-2",
+					},
+					Vendor:     "nvidia",
+					Device:     "tesla-v105",
+					ActivePIDs: []int{185},
+				},
+			},
+			expectedTagMap: map[int32][]string{
+				1234: {
+					"gpu_uuid:gpu-1",
+					"gpu_device:tesla-v100",
+					"gpu_vendor:nvidia",
+				},
+				185: {
+					"gpu_uuid:gpu-2",
+					"gpu_device:tesla-v105",
+					"gpu_device:tesla-v100",
+					"gpu_uuid:gpu-1",
+					"gpu_vendor:nvidia",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup mocked dependencies and ProcessCheck
+			mockWmeta := fxutil.Test[workloadmetamock.Mock](t, fx.Options(
+				core.MockBundle(),
+				fx.Supply(context.Background()),
+				workloadmetafxmock.MockModule(workloadmeta.NewParams()),
+			))
+
+			fakeTagger := taggerMock.SetupFakeTagger(t)
+			gpuDetector := NewGPUSubscriber(mockWmeta, fakeTagger)
+			gpuDetector.SetGPUDetected(tt.detectedGPU)
+
+			// Populate workloadmeta and tagger stores with mocked data
+			for _, gpu := range tt.gpus {
+				mockWmeta.Set(&gpu)
+				fakeTagger.SetTags(types.NewEntityID(types.GPU, gpu.ID), "fake", []string{"gpu_uuid:" + gpu.ID, "gpu_device:" + gpu.Device, "gpu_vendor:" + gpu.Vendor}, nil, nil, nil)
+			}
+
+			actualTagMap := gpuDetector.GetGPUTags()
+			assert.Equal(t, len(tt.expectedTagMap), len(actualTagMap))
+			for pid, tagMap := range tt.expectedTagMap {
+				assert.ElementsMatch(t, actualTagMap[pid], tagMap)
+			}
```

Comment:
why not just this? 
```suggestion
			assert.Equal(t, tt.expectedTagMap, actualTagMap)
```

Reply:
Explained in more depth in the comment here: https://github.com/DataDog/datadog-agent/pull/33952#discussion_r1962202625

`assert.Equal()` requires ordering of the tags, and the tags per pid are ordered non-deterministically. This is why `assert.ElementsMatch` is used to ignore order (otherwise we would need to implement some sorting logic of the tags before asserting). 

Replacing the loop with `assert.ElementsMatch(t, tt.expectedTagMap, actualTagMap)` would not work either as ElementsMatch expects array/slice and not map types. That's why I iterate to assert on each tagMap which is of type `string[]`.